app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
+    new_status = significant_status
+    if (previous_status != new_status) # we only care if something changed
```

Comment:
Just checking the `state` isn't good enough. If the old `significant_status` was `failure` and another status comes in that is also `failure` we still need to fire the webhook for the new status. I've added a test for that case.

Also, `ActiveRecord ==` uses the value of the `id` column: http://apidock.com/rails/ActiveRecord/Core/%3D%3D